src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -208,27 +228,35 @@ function useChatUtilsHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatUtilsHook {
       channelData.channelPicture = channelImage;
       channelData.name = channelName;
 
-      queue.addPriorityJob({
-        operationLabel: DatabaseOperationLabel.FetchChannelDetail_SaveChannelList,
-        id: channel?.id,
-        priority: QueueJobPriority.MEDIUM,
-        task: async () => {
-          channelData.saveIfLatest(localDb);
-        }
-      });
-    }
-
-    originalMembers?.map((member) => {
-      const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
-      try {
+      if (Platform.OS === 'android') {
+        channelData.saveIfLatest(localDb);
+      } else {
         queue.addPriorityJob({
-          operationLabel: DatabaseOperationLabel.FetchChannelDetail_SaveUserMember,
-          id: `${channel?.name}-${member?.user?.username}`,
+          operationLabel: DatabaseOperationLabel.FetchChannelDetail_SaveChannelList,
+          id: channel?.id,
           priority: QueueJobPriority.MEDIUM,
           task: async () => {
-            user.saveOrUpdateIfExists(localDb);
+            channelData.saveIfLatest(localDb);
           }
         });
+      }
+    }
+
+    originalMembers?.map((member) => {
+      const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+      try {
+        if (Platform.OS === 'android') {
+          user.saveOrUpdateIfExists(localDb);
+        } else {
+          queue.addPriorityJob({
+            operationLabel: DatabaseOperationLabel.FetchChannelDetail_SaveUserMember,
+            id: `${channel?.name}-${member?.user?.username}`,
+            priority: QueueJobPriority.MEDIUM,
+            task: async () => {
+              user.saveOrUpdateIfExists(localDb);
+            }
+          });
+        }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous hunk, platform-specific conditional checks are introduced for saving channel details and user members. The same issue with error handling applies here as well. It would be better to wrap these operations in try-catch blocks to handle any potential errors.

```diff
      if (Platform.OS === 'android') {
        try {
          channelData.saveIfLatest(localDb);
        } catch (error) {
          console.error('Error saving channel data:', error);
        }
      } else {
         // ... existing code ...
      }
    }

    originalMembers?.map((member) => {
      const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
      try {
        if (Platform.OS === 'android') {
          user.saveOrUpdateIfExists(localDb);
        } else {
          // ... existing code ...
        }
      } catch (e) {
        console.log('error on save user from channel detail fetch', e);
      }
    });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->